lib/src/utils/dart_symbol_map.dart

```diff
@@ -0,0 +1,104 @@
+import 'package:file/file.dart';
+
+import '../configuration.dart';
+import 'flutter_debug_files.dart';
+
+/// If [configuredPath] is provided, validates and returns the absolute path to the Dart symbol map.
+/// If not provided, returns null.
+///
+/// Note: we do not scan the filesystem for this file because the file does not
+/// have a special extension so worst case we would have to check every file.
+Future<String?> resolveDartSymbolMapPath({
```

Comment:
Looking at the implementation, this is rather generic and not at all specific to symbol map files.  Something like `getAbsolutePath(String path)` is more explicit and easier to read.